packages/cli/src/ui/commands/promptCommands.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SlashCommand,
+  SlashCommandActionReturn,
+  CommandContext,
+  CommandKind,
+} from './types.js';
+import {
+  Config,
+  getMCPServerPrompts,
+  getErrorMessage,
+  invokeMcpPrompt,
+  connectToMcpServer,
+} from '@google/gemini-cli-core';
+
+export function createPromptCommands(config: Config | null): SlashCommand[] {
+  const promptCommands: SlashCommand[] = [];
+  if (config) {
+    const mcpServers = config.getMcpServers() || {};
+    for (const serverName in mcpServers) {
+      const prompts = getMCPServerPrompts(serverName) || [];
+      for (const prompt of prompts) {
+        const commandName = `${prompt.name}`;
+        const newPromptCommand: SlashCommand = {
+          name: commandName,
+          description: prompt.description || `Invoke prompt ${prompt.name}`,
+          kind: CommandKind.BUILT_IN,
+          subCommands: [
+            {
+              name: 'help',
+              description: 'Show help for this prompt',
+              kind: CommandKind.BUILT_IN,
+              action: async (): Promise<SlashCommandActionReturn> => {
+                if (!prompt.arguments || prompt.arguments.length === 0) {
+                  return {
+                    type: 'message',
+                    messageType: 'info',
+                    content: `Prompt "${prompt.name}" has no arguments.`,
+                  };
+                }
+
+                let helpMessage = `Arguments for "${prompt.name}":\n\n`;
+                for (const arg of prompt.arguments) {
+                  helpMessage += `  --${arg.name}\n`;
+                  if (arg.description) {
+                    helpMessage += `    ${arg.description}\n`;
+                  }
+                  helpMessage += `    (required: ${arg.required ? 'yes' : 'no'})\n\n`;
+                }
+                return {
+                  type: 'message',
+                  messageType: 'info',
+                  content: helpMessage,
+                };
+              },
+            },
+          ],
+          action: async (
+            context: CommandContext,
+            args: string,
+          ): Promise<SlashCommandActionReturn> => {
+            const { config } = context.services;
+            if (!config) {
+              return {
+                type: 'message',
+                messageType: 'error',
+                content: 'Config not loaded.',
+              };
+            }
+
+            const promptArgs = args.trim();
+
+            const argValues: { [key: string]: string } = {};
+            // arg parsing: --key="value" or --key=value
+            const argRegex = /--([^=]+)=(?:"((?:\\.|[^"\\])*)"|([^ ]*))/g;
```

Comment:
Do other clients also try and parse args in this way? It does feel a bit heavy to provide name => value. I would have imagined that other clients would allow positional arguments in some form. Would you mind double checking this bit?